synth_node_bin/src/main.rs

```diff
@@ -0,0 +1,125 @@
+//! A synthetic node binary can be used to interact with the node in the
+//! background from a different runtime environment.
+//!
+//! This is a quick solution - in future Ziggurat projects it will be done differently.
+//!
+//! Remaining worklist (probably not going to be implemented in this repo):
+//!   - Add an argument option to choose specific action in order to support
+//!     different synthetic node binary implementations at once. A few examples:
+//!     ```
+//!        ./synthetic_node_bin --action=A    // Runs an idle/friendly synthetic node
+//!        ./synthetic_node_bin --action=B    // Runs a wild synthetic node which does something funny
+//!     ```
+use std::{net::SocketAddr, process::ExitCode};
+
+use anyhow::Result;
+use clap::Parser;
+use tokio::time::{sleep, Duration};
+use ziggurat_zcash::{protocol::message::Message, tools::synthetic_node::SyntheticNode};
+
+/// A synthetic node which can connect to the XRPL node and preform some
+/// actions independently.
+#[derive(Parser)]
+#[command(author, version, about, long_about = None)]
+struct CmdArgs {
+    /// An address of the node in the <ip>:<port> format.
+    #[arg(short = 'i', long)]
+    node_addr: Option<SocketAddr>,
+
+    /// Always reconnect in the case the connection fails - synthetic node never dies.
+    #[arg(short = 's', long, default_value_t = false)]
+    stubborn: bool,
+
+    /// Enable tracing.
+    #[arg(short = 't', long, default_value_t = false)]
+    tracing: bool,
+}
+
+#[tokio::main]
+async fn main() -> ExitCode {
+    let args = CmdArgs::parse();
+
+    let node_addr = if let Some(addr) = args.node_addr {
+        addr
+    } else {
+        eprintln!("Node address should be provided.");
+        return ExitCode::FAILURE;
+    };
+
+    if args.tracing {
+        println!("Enabling tracing.");
+        use tracing_subscriber::{fmt, EnvFilter};
+
+        fmt()
+            .with_test_writer()
+            .with_env_filter(EnvFilter::from_default_env())
+            .init();
+    }
+
+    loop {
+        println!("Starting a synthetic node.");
+
+        if let Err(e) = start_synth_node(node_addr).await {
```

Comment:
I'd call this `run_synth_node` or `do_synth_node` -- you're doing many more things here than starting a synth node.

Reply:
Good point! Thank you!

Reply:
Done